src/contracts/middleware/BLSPubkeyRegistry.sol

```diff
@@ -0,0 +1,211 @@
+
+        emit DeregistrationEvent(operator, pubkeyHash, globalApkHash);
+    }
+
+    /// @notice returns the `ApkUpdate` struct at `index` in the list of APK updates for the `quorumNumber`
+    function getApkUpdateForQuorumByIndex(uint8 quorumNumber, uint256 index) external view returns (ApkUpdate memory){
+        return quorumToApkUpdates[quorumNumber][index];
+    }
+
+    /// @notice returns the `ApkUpdate` struct at `index` in the list of APK updates that keep track of the sum of the APK amonng all quorums
+    function globalApkUpdates(uint256 index) external view returns (ApkUpdate memory){
+        return globalApkUpdateList[index];
+    }
+
+    function quorumApk(uint8 quorumNumber) external view returns (BN254.G1Point memory){
+        return quorumToApk[quorumNumber];
+    }
+
+    function globalApk() external view returns (BN254.G1Point memory){
+        return _globalApk;
+    }
+
+
+    /**
+     * @notice get hash of the apk of `quorumNumber` at `blockNumber` using the provided `index`;
+     * called by checkSignatures in BLSSignatureChecker.sol.
+     */
+    function getApkHashForQuorumAtBlockNumberFromIndex(uint8 quorumNumber, uint32 blockNumber, uint256 index) external view returns (bytes32){
+        ApkUpdate memory quorumApkUpdate = quorumToApkUpdates[quorumNumber][index];
+        require(blockNumber >= quorumApkUpdate.updateBlockNumber, "BLSPubkeyRegistry.getApkHashForQuorumAtBlockNumberFromIndex: index too recent");
+
+        if (index != quorumToApkUpdates[quorumNumber].length - 1){
+            require(blockNumber < quorumApkUpdate.nextUpdateBlockNumber, "BLSPubkeyRegistry.getApkHashForQuorumAtBlockNumberFromIndex: not latest apk update");
+        }
+
+        return quorumApkUpdate.apkHash;
+    }
+
+	/**
+     * @notice get hash of the apk among all quourums at `blockNumber` using the provided `index`;
+     * called by checkSignatures in BLSSignatureChecker.sol.
+     */
+    function getGlobalApkHashAtBlockNumberFromIndex(uint32 blockNumber, uint256 index) external view returns (bytes32){
+        ApkUpdate memory globalApkUpdate = globalApkUpdateList[index];
+        require(blockNumber >= globalApkUpdate.updateBlockNumber, "BLSPubkeyRegistry.getGlobalApkHashAtBlockNumberFromIndex: blockNumber too recent");
+
+        if (index != globalApkUpdateList.length - 1){
+            require(blockNumber < globalApkUpdate.nextUpdateBlockNumber, "getGlobalApkHashAtBlockNumberFromIndex.getGlobalApkHashAtBlockNumberFromIndex: not latest apk update");
+        }
+
+        return globalApkUpdate.apkHash;
+    }
+
+    function getQuorumApkHistoryLength(uint8 quorumNumber) external view returns(uint32){
+        return uint32(quorumToApkUpdates[quorumNumber].length);
+    }
+
+    function getGlobalApkHistoryLength() external view returns(uint32){
+        return uint32(globalApkUpdateList.length);
+    }
+
+    function _processGlobalApkUpdate(BN254.G1Point memory newGlobalApk) internal returns(bytes32){
```

Comment:
i find it a bit weird that this param is the new apk becuase we don't do that in the quorum function? inputting the point to add to the globalApkBeforeUpdate is cleaner. See https://github.com/Layr-Labs/eigenlayer-contracts/pull/47/files#r1207099006